src/mongo/delegates/Updates.class.php

```diff
@@ -467,10 +468,10 @@ protected function applyChangeSet(\Tripod\ChangeSet $cs, $originalCBDs, $context
 
                 $doc = $this->getDocumentForUpdate($subjectOfChange, $contextAlias, $originalCBDs);
 
-                $targetGraph = new MongoGraph();
-                $targetGraph->add_tripod_array($doc);
+//                $targetGraph = new MongoGraph();
+//                $targetGraph->add_tripod_array($doc);
 
-                if(!$targetGraph->has_triples_about($subjectOfChange))
+                if(empty($doc))
                 {
                     $upsert = true;
```

Comment:
Yes, but ` $doc ` for the next ` $change ` might *not* be empty - we never set ` $upsert ` back to false, so if the first one is new, it will treat all the other documents as new.